payroll/views.py

```diff
@@ -240,9 +230,14 @@ def composition_data(self):
         composition_json = []
         percentage_tracker = 0
 
+        unit_base_pay, unit_extra_pay = super().get_entity_payroll()
+        budget = unit_base_pay + unit_extra_pay
+        for dept in top_departments:
```

Comment:
Since you're looping over `top_departments` below, how about performing this calculation in that loop?